### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by briandowns** [Jason2605/Dictu#662] [General]
Do you have a roadmap on features, bug fixes, and/or capabilities you'd like to see out of Dictu in the short or long term?
Was curious if there's any specific audience you wanted to cater to? ie general, cloud, etc.
Do we want to publish dev binaries?

Hackweek at Suse is coming up next month and was curious about this stuff to possibly get some involvement.

**Comment by Jason2605** [comment]
It's a great question and one I'm quite open to at the same time. Obviously any of the open issues are welcome additions, but I think the big big **big** glaring one for me is getting Fibers into the language - although this one is also a pretty substantial chunk of work. An ability for some sort of concurrency I think is really holding Dictu back. Another one is dealing with strings, we can get the changes you talked about for the hex and octal values in so we can use escape codes, and then potentially going one further and having actual UTF8 strings and probably an additional bytes type.

I then think there are alternative contributions for people who may not want to dive head deep into the C code base, for example like improving the external tools (VSc plugin, maybe create a linter etc).

As for audience, again it's a good question, I think for the most part i've been thinking of it as a pretty general base as there isn't really a niche that Dictu fills in all honesty, probably not a great answer when thinking of additions, sorry!

As always though I'm def open to any ideas you or anyone else has!

**Reply by Jason2605** [reply]
Another point actually I've just though of is that the runtime / compile time errors at times can be very cryptic, I think it would be nice to have a think about these and potentially clean this up a bit

**Comment by briandowns** [comment]
Apologies on getting back to a reply on this thread. I don't know how this slipped by. I think Fibers is, sadly, a bit above my head in terms of implementation. However, I'd like to take the implementation of a bytes type on. Any thoughts or guidance on that would be helpful.

**Reply by Jason2605** [reply]
It’s all good! I think Fibers will be a mammoth task to be honest as it’s probably going to impact pretty much everything haha, but at least we have Wren to go off of as a base. Yeah be great if you wanted to pickup the bytes work! The bytes type itself probably won’t be too different to what our string type is currently I’d imagine with most of the work being implementing the UTF-8 library

**Reply by briandowns** [reply]
Would you want it an importable module or part of the language? Eg b'sometbing'?

**Reply by Jason2605** [reply]
I’d have it as part of the language yeah, and yeah I’d follow python with b”” in the same way we have r”” (also borrowed from Python)
